packages/expo-screen-orientation/plugin/src/withScreenOrientation.ts

```diff
@@ -20,54 +17,52 @@ const OrientationLock = {
   LANDSCAPE_RIGHT: 'UIInterfaceOrientationMaskLandscapeRight',
 };
 
-export function modifyObjcAppDelegate(contents: string, mask: string): string {
-  // Add import
-  if (!contents.includes('#import <EXScreenOrientation/EXScreenOrientationViewController.h>')) {
-    contents = contents.replace(
-      /#import "AppDelegate.h"/g,
-      `#import "AppDelegate.h"
-#import <EXScreenOrientation/EXScreenOrientationViewController.h>`
-    );
-  }
+type OrientationMasks = keyof typeof OrientationLock;
 
-  // Change View Controller
-  if (!contents.includes('[EXScreenOrientationViewController alloc]')) {
-    contents = contents.replace(
-      /UIViewController\s?\*\s?rootViewController\s?=\s?\[UIViewController new\];/g,
-      `UIViewController *rootViewController = [[EXScreenOrientationViewController alloc] initWithDefaultScreenOrientationMask:${mask}];`
-    );
-  }
-  return contents;
+// `initialOrientation` is not public in expo config yet, we just use it as an internal type.
+interface ExpoConfigWithInitialOrientation extends ExpoConfig {
+  initialOrientation?: OrientationMasks;
 }
 
 const withScreenOrientationViewController: ConfigPlugin<{
   initialOrientation?: keyof typeof OrientationLock;
 } | void> = (config, { initialOrientation = 'DEFAULT' } = {}) => {
+  config = withInfoPlist(config, (config) => {
+    const extendedConfig = {
+      ...config,
+      initialOrientation,
+    };
+    config.modResults = setInitialOrientation(extendedConfig, config.modResults);
+    return config;
+  });
+  return config;
+};
+
+export function getInitialOrientation(
+  config: Pick<ExpoConfigWithInitialOrientation, 'initialOrientation'>
+): OrientationMasks {
+  return config.initialOrientation ?? 'DEFAULT';
```

Comment:
Any particular reason we're using `DEFAULT` instead of undefined, the actual default value would be inferred in Objective C anyways?